clang/test/SemaHLSL/Availability/avail-diag-strict-lib.hlsl

```diff
@@ -129,6 +129,55 @@ class MyClass
   }
 };
 
+// Exported function without body - not used
+export void exportedFunctionUnused(float f);

```

Comment:
Will there be an error if has export when declare like
```
export void export_mismatch(float f);
```
But without export when define like
```
void export_mismatch(float f) { return f; }
```

Reply:
Good catch! This is currently not reporting an error and the redeclaration is not recognized as exported function and therefore not included in the diagnostic scan. I believe the check whether to scan a function or not should be based on whether the function has an external linkage or not. This cannot be implemented until issue #92071 is resolved though. I'll see if I can fix this without the default linkage change.

Reply:
Done.

Reply:
Also adding test for this here: https://github.com/llvm/llvm-project/pull/97370